ocp_utilities/infra.py

```diff
@@ -354,3 +357,69 @@ def create_icsp(icsp_name, repository_digest_mirrors):
     )
     icsp.deploy()
     return icsp
+
+
+def encode_secret_data(secret_data_dict):
+    """Encoding secret data in base64
+
+    Args:
+        secret_data_dict (dict): Secret data to be encoded
+    Returns:
+        secret_data_encoded (str): secret_data_dict encoded in base64
+    """
+    return base64.b64encode(json.dumps(secret_data_dict).encode("ascii")).decode(
+        "utf-8"
+    )
+
+
+def create_update_pull_secret(secret_data_dict, name, namespace):
+    """Updating existing pull-secret / Creating new pull-secret
```

Comment:
```suggestion
    """Update existing secret or create a new secret; secret type - dockerconfigjson
```
please remove "pull" - this is not relevant. you work with `secret`

Reply:
done